# <font color='blue'>Data Science Academy</font>

## <font color='blue'>Fundamentos de Linguagem Python Para Análise de Dados e Data Science</font>

## <font color='blue'>Projeto 2</font>

## <font color='blue'>Análise Exploratória de Dados em Linguagem Python Para a Área de Varejo</font>

![DSA](imagens/projeto2.png)

In [1]:
# Versão da Linguagem Python
from platform import python_version
print('Versão da Linguagem Python Usada Neste Jupyter Notebook:', python_version())

Versão da Linguagem Python Usada Neste Jupyter Notebook: 3.9.13


In [2]:
# Imports
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import datetime as dt

## Carregando os Dados

In [3]:
# Carrega o dataset
df_dsa = pd.read_csv('dados/dataset.csv')

In [4]:
# Shape
df_dsa.shape

(9700, 11)

In [5]:
# Amostra dos dados
df_dsa.head()

ID_Pedido Data_Pedido ID_Cliente   Segmento           Pais  \
0  CA-2017-152156  08/11/2017   CG-12520   Consumer  United States   
1  CA-2017-152156  08/11/2017   CG-12520   Consumer  United States   
2  CA-2017-138688  12/06/2017   DV-13045  Corporate  United States   
3  US-2016-108966  11/10/2016   SO-20335   Consumer  United States   
4  US-2016-108966  11/10/2016   SO-20335   Consumer  United States   

            Cidade      Estado       ID_Produto        Categoria SubCategoria  \
0        Henderson    Kentucky  FUR-BO-10001798        Furniture    Bookcases   
1        Henderson    Kentucky  FUR-CH-10000454        Furniture       Chairs   
2      Los Angeles  California  OFF-LA-10000240  Office Supplies       Labels   
3  Fort Lauderdale     Florida  FUR-TA-10000577        Furniture       Tables   
4  Fort Lauderdale     Florida  OFF-ST-10000760  Office Supplies      Storage   

   Valor_Venda  
0     261.9600  
1     731.9400  
2      14.6200  
3     957.5775  
4      22.3680

In [6]:
# Amostra dos dados
df_dsa.tail()

ID_Pedido Data_Pedido ID_Cliente     Segmento           Pais  \
9695  CA-2018-154116  15/12/2018   KM-16660     Consumer  United States   
9696  CA-2018-154116  15/12/2018   KM-16660     Consumer  United States   
9697  CA-2018-154116  15/12/2018   KM-16660     Consumer  United States   
9698  CA-2017-105291  30/10/2017   SP-20920     Consumer  United States   
9699  CA-2018-147032  31/07/2018   LB-16795  Home Office  United States   

               Cidade      Estado       ID_Produto        Categoria  \
9695        Inglewood  California  OFF-PA-10004569  Office Supplies   
9696        Inglewood  California  OFF-AP-10000027  Office Supplies   
9697        Inglewood  California  TEC-PH-10000675       Technology   
9698  San Luis Obispo  California  OFF-FA-10003059  Office Supplies   
9699       Wilmington    Delaware  OFF-PA-10003256  Office Supplies   

     SubCategoria  Valor_Venda  
9695        Paper       22.830  
9696   Appliances       54.320  
9697       Phones      196.776  
9698    Fasteners        3.620  
9699        Paper       11.540

## Análise Exploratória

In [7]:
# Colunas do conjunto de dados
df_dsa.columns

Index(['ID_Pedido', 'Data_Pedido', 'ID_Cliente', 'Segmento', 'Pais', 'Cidade',
       'Estado', 'ID_Produto', 'Categoria', 'SubCategoria', 'Valor_Venda'],
      dtype='object')

In [8]:
# Verificando o tipo de dado de cada coluna
df_dsa.dtypes

ID_Pedido        object
Data_Pedido      object
ID_Cliente       object
Segmento         object
Pais             object
Cidade           object
Estado           object
ID_Produto       object
Categoria        object
SubCategoria     object
Valor_Venda     float64
dtype: object

In [9]:
# Resumo estatístico da coluna com o valor de venda
df_dsa['Valor_Venda'].describe()

count     9700.000000
mean       230.469892
std        627.504252
min          0.444000
25%         17.248000
50%         54.272000
75%        209.932500
max      22638.480000
Name: Valor_Venda, dtype: float64

In [10]:
# Verificando se há registros duplicados
df_dsa[df_dsa.duplicated()]

Empty DataFrame
Columns: [ID_Pedido, Data_Pedido, ID_Cliente, Segmento, Pais, Cidade, Estado, ID_Produto, Categoria, SubCategoria, Valor_Venda]
Index: []

In [11]:
# Verificando de há valores ausentes
df_dsa.isnull().sum()

ID_Pedido       0
Data_Pedido     0
ID_Cliente      0
Segmento        0
Pais            0
Cidade          0
Estado          0
ID_Produto      0
Categoria       0
SubCategoria    0
Valor_Venda     0
dtype: int64

In [12]:
df_dsa.head()

ID_Pedido Data_Pedido ID_Cliente   Segmento           Pais  \
0  CA-2017-152156  08/11/2017   CG-12520   Consumer  United States   
1  CA-2017-152156  08/11/2017   CG-12520   Consumer  United States   
2  CA-2017-138688  12/06/2017   DV-13045  Corporate  United States   
3  US-2016-108966  11/10/2016   SO-20335   Consumer  United States   
4  US-2016-108966  11/10/2016   SO-20335   Consumer  United States   

            Cidade      Estado       ID_Produto        Categoria SubCategoria  \
0        Henderson    Kentucky  FUR-BO-10001798        Furniture    Bookcases   
1        Henderson    Kentucky  FUR-CH-10000454        Furniture       Chairs   
2      Los Angeles  California  OFF-LA-10000240  Office Supplies       Labels   
3  Fort Lauderdale     Florida  FUR-TA-10000577        Furniture       Tables   
4  Fort Lauderdale     Florida  OFF-ST-10000760  Office Supplies      Storage   

   Valor_Venda  
0     261.9600  
1     731.9400  
2      14.6200  
3     957.5775  
4      22.3680

## Pergunta de Negócio 1:

### Qual Cidade com Maior Valor de Venda de Produtos da Categoria 'Office Supplies'?

## Pergunta de Negócio 2:

### Qual o Total de Vendas Por Data do Pedido?

Demonstre o resultado através de um gráfico de barras.

## Pergunta de Negócio 3:

### Qual o Total de Vendas por Estado?

Demonstre o resultado através de um gráfico de barras.

## Pergunta de Negócio 4:

### Quais São as 10 Cidades com Maior Total de Vendas?

Demonstre o resultado através de um gráfico de barras.

## Pergunta de Negócio 5:

### Qual Segmento Teve o Maior Total de Vendas?

Demonstre o resultado através de um gráfico de pizza.

## Pergunta de Negócio 6 (Desafio Nível Baby):

### Qual o Total de Vendas Por Segmento e Por Ano?

## Pergunta de Negócio 7 (Desafio Nível Júnior):

Os gestores da empresa estão considerando conceder diferentes faixas de descontos e gostariam de fazer uma simulação com base na regra abaixo:

- Se o Valor_Venda for maior que 1000 recebe 15% de desconto.
- Se o Valor_Venda for menor que 1000 recebe 10% de desconto.

### Quantas Vendas Receberiam 15% de Desconto?

## Pergunta de Negócio 8 (Desafio Nível Master):

### Considere Que a Empresa Decida Conceder o Desconto de 15% do Item Anterior. Qual Seria a Média do Valor de Venda Antes e Depois do Desconto?

## Pergunta de Negócio 9 (Desafio Nível Master Ninja):

### Qual o Média de Vendas Por Segmento, Por Ano e Por Mês?

Demonstre o resultado através de gráfico de linha.

## Pergunta de Negócio 10 (Desafio Nível Master Ninja das Galáxias):

### Qual o Total de Vendas Por Categoria e SubCategoria, Considerando Somente as Top 12 SubCategorias? 

Demonstre tudo através de um único gráfico.